navigation/AppContainer.tsx

```diff
@@ -2,19 +2,65 @@ import * as React from 'react';
 import {NavigationContainer} from '@react-navigation/native';
 import createStackNavigator from './createStackNavigator';
 import {RootStackParamList} from './type';
-import Text from '../components/Text';
+import {AppState} from 'react-native';
 import Intro from './Intro';
 import CreateNewWallet from './CreateNewWallet';
-import {View} from 'react-native';
 import Loading from '../components/Loading';
 import OnBoarding from '../screens/OnBoarding';
 import Wallet from '../screens/Wallet';
 import OpenWallet from './OpenWallet';
 import ImportWallet from './ImportWallet';
+import useLockedScreen from '../hooks/useLockedScreen';
+import useAsyncStorage from '../hooks/useAsyncStorage';
+import useWallet from '../hooks/useWallet';
+import {useEffect, useRef, useState} from 'react';
+import LocalAuth from '../utils/LocalAuth';
+import {BlurView} from 'expo';
 
 const Stack = createStackNavigator<RootStackParamList>();
 
 const AppContainer = (props: any) => {
+  const {lockedScreen, setLockedScreen} = useLockedScreen();
+  const [lockAfterBackground, setLockAfterBackground] = useAsyncStorage(
+    'lockAfterBackground',
+    'false',
+  );
+
+  const {refreshWallet} = useWallet();
+
+  const appState = useRef(AppState.currentState);
+
+  const [appStateVisible, setAppStateVisible] = useState(appState.current);
+
+  useEffect(() => {
+    if (appStateVisible == 'active' && lockedScreen) {
+      LocalAuth((error: any) => {
+        if (!error) {
+          setLockedScreen(false);
+        } else {
+          setLockedScreen(true);
+        }
+      });
+    }
+  }, [appStateVisible]);
```

Comment:
What is the use of setLockedScreen and lockedScreen? I don't see it being used in any other parts of the codebase (Searching for `useLockedScreen` across the codebase)

Reply:
in this same file, line 52, lockedScreen is set to `true` if the wallet setting is set to lock the app after coming back from background, which triggers the biometrics check